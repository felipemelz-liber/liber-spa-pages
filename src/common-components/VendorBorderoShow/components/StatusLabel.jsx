import React from 'react';
import PropTypes from 'prop-types';
import { Container, WhiteCircle } from './StatusLabel.styles';

const StatusLabel = ({ children, status, color }) => {
  return (
    <>
      <Container status={status} color={color}>
        <WhiteCircle />
        {children}
      </Container>
    </>
  );
};

StatusLabel.propTypes = {
  children: PropTypes.node,
  status: PropTypes.oneOf(['positive', 'negative', 'neutral', 'primary']),
  color: PropTypes.string,
};

StatusLabel.defaultProps = {
  children: '',
  status: 'neutral',
  color: '',
};

export default StatusLabel;
