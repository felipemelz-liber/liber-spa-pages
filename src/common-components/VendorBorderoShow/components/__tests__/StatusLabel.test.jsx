import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import StatusLabel from '../StatusLabel';

describe('StatusLabel component tests', () => {
  configure({ adapter: new Adapter() });

  it('should render correctly without props', () => {
    const wrapper = shallow(<StatusLabel />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  ['positive', 'negative', 'neutral', 'primary'].forEach(status => {
    it(`should render correctly with ${status} status`, () => {
      const wrapper = shallow(<StatusLabel status={status}>Mock</StatusLabel>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
