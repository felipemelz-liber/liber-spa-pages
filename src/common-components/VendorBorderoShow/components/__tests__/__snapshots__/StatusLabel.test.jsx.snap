// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusLabel component tests should render correctly with negative status 1`] = `
<Fragment>
  <styled.div
    color=""
    status="negative"
  >
    <styled.div />
    Mock
  </styled.div>
</Fragment>
`;

exports[`StatusLabel component tests should render correctly with neutral status 1`] = `
<Fragment>
  <styled.div
    color=""
    status="neutral"
  >
    <styled.div />
    Mock
  </styled.div>
</Fragment>
`;

exports[`StatusLabel component tests should render correctly with positive status 1`] = `
<Fragment>
  <styled.div
    color=""
    status="positive"
  >
    <styled.div />
    Mock
  </styled.div>
</Fragment>
`;

exports[`StatusLabel component tests should render correctly with primary status 1`] = `
<Fragment>
  <styled.div
    color=""
    status="primary"
  >
    <styled.div />
    Mock
  </styled.div>
</Fragment>
`;

exports[`StatusLabel component tests should render correctly without props 1`] = `
<Fragment>
  <styled.div
    color=""
    status="neutral"
  >
    <styled.div />
  </styled.div>
</Fragment>
`;
